cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/cloudsqlpostgresql/BQValidation.java

```diff
@@ -0,0 +1,268 @@
+     *
+     * @param rsSource     The result set obtained from the Oracle database.
+     * @param bigQueryData The list of BigQuery JSON objects to compare with the result set data.
+     * @return True if the result set data matches the BigQuery data, false otherwise.
+     * @throws SQLException   If an SQL error occurs during the result set operations.
+     * @throws ParseException If an error occurs while parsing the data.
+     */
+    public static boolean compareResultSetandJsonData(ResultSet rsSource, List<JsonObject> bigQueryData)
+            throws SQLException {
+        ResultSetMetaData mdSource = rsSource.getMetaData();
+        boolean result = false;
+        int columnCountSource = mdSource.getColumnCount();
+
+        if (bigQueryData == null) {
+            Assert.fail("bigQueryData is null");
+            return result;
+        }
+        // Get the column count of the first JsonObject in bigQueryData
+        int jsonObjectIdx = 0;
+        int columnCountTarget = 0;
+        if (bigQueryData.size() > 0) {
+            columnCountTarget = bigQueryData.get(jsonObjectIdx).entrySet().size();
+        }
+        // Compare the number of columns in the source and target
+        Assert.assertEquals("Number of columns in source and target are not equal",
+                columnCountSource, columnCountTarget);
+
+        while (rsSource.next()) {
+            int currentColumnCount = 1;
+            while (currentColumnCount <= columnCountSource) {
+                String columnTypeName = mdSource.getColumnTypeName(currentColumnCount);
+                int columnType = mdSource.getColumnType(currentColumnCount);
+                String columnName = mdSource.getColumnName(currentColumnCount);
+                // Perform different comparisons based on column type
+                switch (columnType) {
+                    case Types.BIT:
+                        boolean bqDateString = bigQueryData.get(jsonObjectIdx).get(columnName).getAsBoolean();
+                        result = getBooleanValidation(rsSource, String.valueOf(bqDateString), columnName, columnTypeName);
+                        Assert.assertTrue("Different values found for column : %s", result);
+                        break;
+
+                    case Types.DECIMAL:
+                    case Types.NUMERIC:
+                        BigDecimal sourceDecimal = rsSource.getBigDecimal(currentColumnCount);
+                        BigDecimal targetDecimal = bigQueryData.get(jsonObjectIdx).get(columnName).getAsBigDecimal();
+                        int desiredScale = 2; // Set the desired scale (number of decimal places)
+                        BigDecimal adjustedSourceValue = sourceDecimal.setScale(desiredScale, RoundingMode.HALF_UP);
+                        BigDecimal adjustedTargetValue = targetDecimal.setScale(desiredScale, RoundingMode.HALF_UP);
```

Comment:
I see we are setting scale here, was there any discrepancy in the original data which was needed to be handled.

Reply:
We need to match precision scale of both side to avoid errors cause due to auto round off when precision mismatch

example -> 2.126 (3 decimal precision) will be converted to 2.13 (2 decimal precision) (round off)
:+1: